### PR TITLE
refactor: global token bar

### PR DESCRIPTION
**Pull request by andreabadesso**
### Acceptance Criteria
- We should refactor the TokenBar component to be a Function component
- We should refactor the TokenBar to be global and render only on specific routes

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -24,6 +25,7 @@ ReactDOM.render(
   <Provider store={store}>
     <GlobalModal>
       <Router>
+        <TokenBar />
```
Comment (andreabadesso) [inline src/index.js:28]: This needs to be rendered under the `Router` provider so we can get the `location` and `history` objects from the `useLocation` and `useHistory` hooks

File: src/components/TokenBar.js [hunk 2]
```diff
@@ -5,220 +5,204 @@
...
-    this.state = {
-      opened: false,
-      selected: 'HTR',
-    }
+import { useHistory, useLocation } from 'react-router-dom';
+
+// Routes that should display the TokenBar
+const ROUTE_WHITELIST = [
+  '/unknown_tokens/',
+  '/wallet/',
```
Comment (tuliomir) [inline src/components/TokenBar.js:23]: This won't match with `/wallet/send_tokens`, right?

**Review by tuliomir — APPROVED** [review]

File: src/components/TokenBar.js [hunk 3]
```diff
@@ -5,220 +5,204 @@
...
+export default function TokenBar () {
+  const [opened, setOpened] = useState(false);
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const location = useLocation();
+
+  const tokens = useSelector((state) => state.tokens);
+  const allTokens = useSelector((state) => state.allTokens);
+  const selectedToken = useSelector((state) => state.selectedToken);
+  const tokensBalance= useSelector((state) => state.tokensBalance);
```
Comment (pedroferreira1) [inline src/components/TokenBar.js:35]:
```suggestion
  const tokensBalance = useSelector((state) => state.tokensBalance);
```

**Review by pedroferreira1 — APPROVED** [review]

File: src/components/TokenBar.js (hunk 2 above)
Comment (andreabadesso) [reply src/components/TokenBar.js:23]: It won't!

File: src/components/TokenBar.js [hunk 4]
```diff
@@ -5,220 +5,204 @@
...
+  '/wallet/',
+];
+
+export default function TokenBar () {
+  const [opened, setOpened] = useState(false);
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const location = useLocation();
+
+  const tokens = useSelector((state) => state.tokens);
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply src/components/TokenBar.js:32]: Yes! Done in c80e7e678367c921e152938981b51374c2da173e

**Pull request merged**